src/ProxyClient/HttpProxyClient.php

```diff
@@ -109,54 +109,33 @@ protected function queueRequest($method, $url, array $headers, $validateHost = t
     protected function escapeTags(array $tags)
     {
         array_walk($tags, function (&$tag) {
-            $tag = str_replace([',', "\n"], ['_', '_'], $tag);
+            $tag = str_replace([',', "\n"], '_', $tag);
         });
 
         return $tags;
     }
 
     /**
-     * Splits a header value into an array of values.
+     * Calculate how many tags fit into the header.
      *
-     * E.g. useful for tag invalidation requests where you might need multiple
-     * requests if you want to invalidate many cache tags to avoid problems
-     * with too long HTTP headers.
+     * This assumes that the tags are separated by one character.
      *
-     * @param string $value
-     * @param int    $length
-     * @param string $delimiter
+     * @param string[] $escapedTags
      *
-     * @return array
+     * @return int Number of tags per tag invalidation request
      */
-    protected function splitLongHeaderValue($value, $length = 7500, $delimiter = ',')
+    protected function determineTagsPerHeader($escapedTags)
     {
-        if (mb_strlen($value) <= $length) {
-            return [$value];
+        if (mb_strlen(implode(',', $escapedTags)) < $this->options['header_length']) {
```

Comment:
Either I'm blind or my comment is gone? Hope I do not repeat myself here now but just in case: I mentioned that `,` is hardcoded now but I might want to use `foobar` as a delimiter later which affects the length of the header so it should be the second argument of this method :)

Reply:
you commented on the commit, instead of in the pull request ;-)

thanks for the input, i guess i should indeed pass in the delimiter, it costs us nothing and is future proof.

Reply:
both xkey and Surrogate-Key (Fastly) works with space separated tags afaik, so a bit outside of this but might be worth to either abstract as stated in this thread or even just align.